src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+            return b;
+        } else if (b == null) {
+            return a;
+        }
+        return Float.sum(a, b);
+    }
+
+    /**
+     * Retrieve basic system stats - auto answering feature, cameras/mics basic info etc.
+     *
+     * @param statistics map to set data to
+     * @throws Exception during http communication, except 403
+     */
+    private void retrieveSystemStatus(Map<String, String> statistics) throws Exception {
+        ArrayNode response = doGet(buildHttpUrl(STATUS), ArrayNode.class);
+        response.iterator().forEachRemaining(jsonNode -> {
+            statistics.put("System Status#" + normalizeStringPropertyName(
+                    getJsonProperty(jsonNode, "langtag").asText().replaceAll("_", " ")),
+                    (getJsonProperty(jsonNode, "stateList")).get(0).asText().replaceAll("_", " ").toUpperCase());
+        });
+    }
+
+    /**
+     * Change property name to start with a capital letter
+     *
+     * @param string parameter name
+     * @return {@link String} property name with first capital letter
+     */
+    private String normalizeStringPropertyName(String string) {
+        return string.substring(0, 1).toUpperCase() + string.substring(1).toLowerCase();
+    }
+
+    /**
+     * Retrieve the current state of the collaboration session
+     *
+     * @return Integer value of the current device volume level
+     * @throws Exception during http communication
+     */
+    private void getCollaborationStatus(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(COLLABORATION), JsonNode.class);
+        if (!response.isNull()) {
+            String sessionState = getJsonProperty(response, "state").asText();
+            statistics.put("Collaboration#Session State", sessionState);
+            if ("ACTIVE".equals(sessionState)) {
+                statistics.put("Collaboration#Session ID", getJsonProperty(response, "id").asText());
+            }
+        }
+    }
+
+    /**
+     * Retrieve current volume level of the device
+     *
+     * @return Integer value of the current device volume level
+     * @throws Exception during http communication
+     */
+    private Integer getVolume() throws Exception {
+        return doGet(buildHttpUrl(VOLUME), Integer.class);
+    }
+
+    /**
+     * Update device's volume with a given value
+     *
+     * @param value target value for the device volume
+     * @return boolean value indicating the success of the operation
+     * @throws Exception during http communication
+     */
+    private boolean setVolume(int value) throws Exception {
+        return doPost(VOLUME, value).getStatusCode().is2xxSuccessful();
+    }
+
+    /**
+     * Retrieve stats of currently active conference call
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    private int listAllConferenceCalls(Map<String, String> statistics) throws Exception {

```

Comment:
Yes, so the thing is that the device itself is able to have multiple conferences at the same time. Since there's no way to figure out which one is the most relevant (also we've had the requirements discussion before and the outcome was that having multiple conferences support is not needed). So basically .get(0) is made based on the consideration that there's at most 1 relevant conference running at the same time. Otherwise - it is an exceptional situation and Im not sure if we should explicitly throw an error in such case since this is not something that would be done using the adapter, however it'll be possible to recover using the adapter (by disconnecting the calls, so if there's an active conference that symphony's aware of - it will be removed, otherwise it will go through all the conferences and will end them)